src/extension/mapml/src/main/java/org/geoserver/mapml/MapMLFeaturesBuilder.java

```diff
@@ -33,21 +35,27 @@ public class MapMLFeaturesBuilder {
     private final GeoServer geoServer;
     private final WMSMapContent mapContent;
     private final GetMapRequest getMapRequest;
+    private final List<Query> queries;
 
     /**
      * Constructor
      *
      * @param mapContent the WMS map content
      * @param geoServer the GeoServer
      */
-    public MapMLFeaturesBuilder(WMSMapContent mapContent, GeoServer geoServer) {
+    public MapMLFeaturesBuilder(
+            WMSMapContent mapContent,
+            GeoServer geoServer,
+            HttpServletRequest httpServletRequest,
+            List<Query> queries) {
```

Comment:
Since the builder supports a single layer I'd recommend simplifying things, and getting a single Query object here, eventually null.

Reply:
fixed